lib/flash.robot

```diff
@@ -92,6 +92,9 @@ Flash Firmware
         Fail    Flash firmware not implemented for platform config ${CONFIG}
     END
 
+    # Don't leave the platform in a state in which it won't react to tests
+    Turn On Power Supply
```

Comment:
If you want DUT to power on after flashing it would be better to use: `Power Cycle On` kwd. As `Turn On Power Suply` does only turn the power on and does not switch the power button by RTE.

Also is there a need for adding this? As I think all of tests which use this kwd and do something to the DUT actually have 'Power On' or similar kwd after flashing?

Reply:
> If you want DUT to power on after flashing it would be better to use: `Power Cycle On` kwd. As `Turn On Power Suply` does only turn the power on and does not switch the power button by RTE.

That was the intent.
 
> Also is there a need for adding this? As I think all of tests which use this kwd and do something to the DUT actually have 'Power On' or similar kwd after flashing?

I don't  know if this is really necessary to have, but as commit message says, this was a problem with running regression tests because `apu-configuration-menu.robot` turned of power supply in teardown for flashing firmware and the tests following it timed out.  Updating `apu-configuration-menu.robot` might be a better idea, I just wanted to avoid this stopping tests multiple times.

Reply:
So this does fix only one test/suite? Its better to edit the test then. This kwd is used in various other tests and adding power on potentially might break those tests.

Reply:
I think that option 1 is the correct one. As it only involves the tests that you had an issue with.

Reply:
Updated that commit: https://github.com/Dasharo/open-source-firmware-validation/pull/518/commits/6d10119365616c62c408a36caa65259e2c8610f9

> I'm thinking that it would be a good idea to add some skip mechanism in suite teardown if whole suite is skipped... To avoid situations like this. And avoid unnecesary flashing of fw when running regression script.

RF seems to be doing it intentionally for cleanup opportunity and there is no way to check that the test was skipped, so it either need to be a flag or a check for whether to skip the tests should be moved into a separate keyword.